tests/helpers/test_script.py

```diff
@@ -4713,7 +4716,7 @@ async def test_shutdown_at(
     expected_trace = {
         "0": [{"result": {"delay": 120.0, "done": False}}],
     }
-    assert_action_trace(expected_trace)
+    assert_action_trace(expected_trace, "cancelled")
```

Comment:
The `await script_obj.async_stop()` works now and the state now correctly reflects as `cancelled`